packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -3,16 +3,55 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames/bind';
 import 'terra-base/lib/baseStyles';
 import Input from 'terra-form/lib/Input';
+import ButtonGroup from 'terra-button-group';
+
 import TimeUtil from './TimeUtil';
 import styles from './TimeInput.scss';
 
 const cx = classNames.bind(styles);
 
+const validDateRegex = new RegExp('^(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]$');
```

Comment:
@benbcai Didn't moment provide a validate date function?

Reply:
Also, I feel like this should be moved to the TimeUtils file.

Reply:
@emilyrohrbough. I think this regex is used to enforce that the hour can only be  between 00 and 23 and the minute be between 00 and 59. Moment does provide an isValid() function to validate that a moment object is initialized with a valid date but I'm not aware of anything for this regex.